editor/src/core/shared/jsx-attributes.ts

```diff
@@ -470,17 +470,30 @@ export function setJSXValueInAttributeAtPath(
           const attributeKey = PP.firstPart(path)
           const tailPath = PP.tail(path)
           const key = `${attributeKey}`
-          const newProps = dropKeyFromNestedObject(attribute, key).content
           if (lastPartOfPath) {
-            return right(
-              jsxAttributeNestedObject(
-                newProps.concat(
-                  jsxPropertyAssignment(key, newAttrib, emptyComments, emptyComments),
+            let updatedExistingProperty = false
+            let updatedContent = attribute.content.map((attr) => {
+              if (attr.type === 'PROPERTY_ASSIGNMENT' && attr.key === key) {
+                updatedExistingProperty = true
+                return jsxPropertyAssignment(key, newAttrib, emptyComments, emptyComments)
```

Comment:
Looks like this will bin off comments if a property is updated. We should probably copy the comments across from the old attribute